File: src/main/scala/nu/tengstrand/tetrisanalyzer/piece/PieceEmpty.scala
package nu.tengstrand.tetrisanalyzer.piece

object PieceEmpty {
  val Number = 0.toByte

  def apply() = new PieceEmpty
}

class PieceEmpty extends Piece {
  val number = PieceEmpty.Number
  val character = '-'
  protected val widths = Array.empty[Int]
  protected val heights = Array.empty[Int]
  protected val shapes = Array.empty[PieceShape]
}

File: langs/java/src/main/java/com/github/tetrisanalyzer/piece/PieceEmpty.java
package com.github.tetrisanalyzer.piece;

import com.github.tetrisanalyzer.settings.PieceSettings;

public class PieceEmpty extends Piece {

    PieceEmpty(PieceSettings settings) {
        super(settings);
    }

    @Override public byte number() { return 0; }
    @Override public char character() { return '-'; }
    @Override protected int[] widths() { return new int[0]; }
    @Override protected int[] heights() { return new int[0]; }
    @Override protected PieceShape[] shapes() { return new PieceShape[0]; }
}